Core/OxidEeEvents.php

```diff
@@ -88,15 +90,34 @@ private static function _insertRowIfNotExists($sTableName, $aKeyValue, $sQuery)
         foreach ($aKeyValue as $sKey => $sValue) {
             $sWhere .= " AND $sKey = '$sValue'";
         }
-
         $sCheckQuery = "SELECT * FROM {$sTableName} WHERE 1" . $sWhere;
         $sExisting = self::$oDb->getOne($sCheckQuery);
-
         if (!$sExisting) {
             self::$oDb->Execute($sQuery);
+            if (self::_isSepaDirectDebit($aKeyValue)) {
```

Comment:
Why is this executed here and not in `_addPaymentMethod`? To me, `_insertRowIfNotExists` is a generic method that shouldn't care about specific payment methods.

If it was executed in `_addPaymentMethod` you would also not need to check `_isSepaDirectDebit`, because it is only executed once anyway.

Reply:
Good point. I moved the execution to `_addPaymentMethod`.

We need `_isSepaDirectDebit` check because we should add SEPA mandate only in the case of SEPA Direct Debit payment method.